torch/testing/_internal/schema_check_tensor.py

```diff
@@ -38,16 +43,69 @@ def __repr__(self):
     @classmethod
     def __torch_dispatch__(cls, func, types, args=(), kwargs=None):
         def unwrap(e):
-            return e.elem if isinstance(e, cls) else e
+            if isinstance(e, cls):
+                return e.elem
+            elif is_iterable_of_dtype(e, cls):
+                return [t.elem for t in e]
+            else:
+                return e
 
         def wrap(e):
-            return cls(e) if isinstance(e, torch.Tensor) else e
+            if isinstance(e, torch.Tensor):
+                return cls(e)
+            elif is_iterable_of_dtype(e, torch.Tensor):
+                return [cls(t) for t in e]
+            else:
+                return e
+
+        def has_mutated(before, after):
+            if isinstance(before, torch.Tensor) and isinstance(after, torch.Tensor):
+                return not torch.equal(before, after)
+            elif is_iterable_of_dtype(before, torch.Tensor) and is_iterable_of_dtype(after, torch.Tensor):
+                before_list = list(before)
+                after_list = list(after)
+                if (len(before_list) != len(after_list)):
+                    return True
+                for before_elem, after_elem in zip(before_list, after_list):
+                    if (not torch.equal(before_elem, after_elem)):
```

Comment:
do we need to handle dicts for any ops?

Reply:
If you mean dictionaries with tensors as values as arguments, I'm not sure if any ops have a signature like that, I can either look into that or just implement a check for that case.